### PR TITLE
Optimizing treasury page

### PR DESCRIPTION
**Pull request by Aliiiu**
# Optimizing Treasury Page
This PR works on optimizing the treasury page

## How should this be tested?

1. Navigate to the treasury page

## Major Bottlenecks I noticed
- API request of static data that can be cached
- potential for parallel API calls
- refetching API calls on navigation

## Method used in resolving bottlenecks
- converted sequential API calls to parallel (wallet details call, token and fraxtalToken calls)
- refactored code to avoid API requests of static data (fetchSfrxETHApr, fetchFraxPairApr)
- refactored code to remove unnecessary use of promise.all() on non-promise calls
- caching the result of the API call for tokens for token and fraxtalToken for 10mins to avoid request at intervals

After

https://github.com/EveripediaNetwork/iq-ui/assets/67792966/ae051034-3f20-42ad-b82e-4c83418c7c37

Before

https://github.com/EveripediaNetwork/iq-ui/assets/67792966/845ac0dd-04fa-4e71-9f75-b909c810c973

## Linked issues
https://github.com/EveripediaNetwork/issues/issues/2601

### CODE REVIEW

**Comment by kesar** [comment]
plz can u elaborate on what are the improvements ? screens that show improvements / explanations

**Comment by Aliiiu** [comment]
> plz can u elaborate on what are the improvements ? screens that show improvements / explanations

Alright sir

**Comment by kesar** [comment]
what should I see in the videos? before looks faster than after :?

**Comment by Aliiiu** [comment]
> what should I see in the videos? before looks faster than after :?

sorry I used the label wrongly

File: src/services/treasury/rest/index.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      baseUrl = '' // Override for FRAX finance API requests
+    } else if (args.startsWith('https://api.thegraph.com')) {
+      baseUrl = '' // Override for The Graph API requests
+    }
+  }
+
+  return fetchBaseQuery({ baseUrl })(args, api, extraOptions)
+}
+
+export const treasuryRestApi = createApi({
```
Comment (SidharthK2) [inline src/services/treasury/rest/index.ts:45]: Can we name it better

File: src/utils/treasury-utils.ts [hunk 2]
```diff
@@ -258,6 +253,7 @@ export const fetchFraxPairApr = async (tokenName: string) => {
       ) /
       10 ** 18
     const apr = interestPerSecond * NUMBER_OF_SECONDS * utilizationRate * 100
+    console.log({ apr })
```
Comment (SidharthK2) [inline src/utils/treasury-utils.ts:256]:
```suggestion
```

**Review by SidharthK2 — APPROVED** [review]

**Review by kesar — APPROVED** [review]

**Pull request merged**